src/Serializers/SnakSerializer.php

```diff
@@ -23,10 +24,17 @@ class SnakSerializer implements DispatchableSerializer {
 	private $dataValueSerializer;
 
 	/**
+	 * @var bool
+	 */
+	private $serializeSnaksWithHash;
+
+	/**
 	 * @param Serializer $dataValueSerializer
+	 * @param bool $serializeSnaksWithHash
 	 */
-	public function __construct( Serializer $dataValueSerializer ) {
+	public function __construct( Serializer $dataValueSerializer, $serializeSnaksWithHash = true ) {
```

Comment:
The variable name is a bit complicated. We already know we are in the SnakSerializer. `$serializeHash` should be enough.